starter-code/routes/authController.js

```diff
@@ -0,0 +1,101 @@
+const express = require('express');
+const bcrypt= require('bcrypt');
+const authController = express.Router();
+
+const User = require('../models/user');
+const bcryptSalt = 10;
+
+authController.get('/signup', (req, res, next) => {
+    res.render('auth/signup')
+})
+
+authController.post('/signup', (req, res, next) => {
+    var username = req.body.username;
+    var name = req.body.name;
+    var email = req.body.email;
+    var password = req.body.password;
+    
+    if (username === "" || password === "" || name === "" || email === "") {
+        res.render("auth/signup", {
+            errorMessage: "Please, fill all the form "
+        })
+        return;
+    }
+    
+    User.findOne({ "username": username },
+    "username",
+    (err, user) => {
+        if (user !== null) {
+            res.render("auth/signup", {
+                errorMessage: "The username already exist !"
+            })
+            return;
+        }
+
+        var salt = bcrypt.genSaltSync(bcryptSalt)
+        var hashPass = bcrypt.hashSync(password, salt);
+
+    var newUser = User({
+        username: username,
+        password: hashPass,
+        name: name,
+        email: email
+
+    })
+
+    newUser.save(err => {
+        if (err) {
+             res.render("auth/signup", {
+               errorMessage: "Oups something went wrong !"
+             });
+        } else {
+            res.render('auth/signup', {
```

Comment:
I think it would be better to redirect the user to the login page here :)

Reply:
Oh yeah you right ! Thanks you ! :D